Added sklearn classification pipeline to main workflow.

dickeygh opened the pull request:
Created new file called `Classifiers.py`, works similarly to `regressors.py`.

Added a new function called classification_metrics (line 316) to `analysis.py`. This function computes and returns several different evaluation metrics of the model's performance on the test set.

Added classification algorithms svc and kneighborsclassifier to the list of models that need feature normalization in `features.py` (line 23).

Changed `main.py` to run either the classification models or the regression models given a user input.

Changed line 17 of `name.py` to fit my directory.

Added a classification_run function to `models.py` that runs the classification models and prints evaluation results.

Changed `ingest.py` line 4 so that drop is not always=true. This is because some of the classification models run better on some data sets without dropping the extra columns. (I made drop=True when running regression models, see `main.py` line 64.)

Added sider data set to dataFiles.

Currently have 4 evaluation methods (accuracy score, confusion matrix, roc auc score, and classification report), 1 data set (sider), and 2 different models (SVC and Kneighborsclassifier). I plan on adding more of all of these.

pep8speaks commented:
Hello @dickeygh! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`core/Classifiers.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/Classifiers.py):

> [Line 14:30](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/Classifiers.py#L14): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`core/analysis.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py):

> [Line 315:1](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py#L315): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 315:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py#L315): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 79 characters)
> [Line 316:1](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py#L316): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 3
> [Line 323:5](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py#L323): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 328:62](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/analysis.py#L328): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file



* In the file [`core/ingest.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/ingest.py):

> [Line 4:1](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/ingest.py#L4): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

* In the file [`core/models.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py):

> [Line 5:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L5): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 116:9](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L116): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 161:5](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L161): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 165:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L165): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (158 > 79 characters)
> [Line 165:93](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L165): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 165:106](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L165): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 171:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L171): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (168 > 79 characters)
> [Line 171:120](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/models.py#L171): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment

* In the file [`core/name.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/name.py):

> [Line 17:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/core/name.py#L17): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 79 characters)

* In the file [`main.py`](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py):

> [Line 15:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L15): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 16:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L16): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 24:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L24): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 26:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L26): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 37:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L37): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 39:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L39): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 48:24](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L48): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 50:78](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L50): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 50:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L50): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (183 > 79 characters)
> [Line 52:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L52): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (183 > 79 characters)
> [Line 52:91](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L52): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 57:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L57): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 63:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L63): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 80:17](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L80): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 80:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L80): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 86:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L86): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 87:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L87): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 87:80](https://github.com/Dlux804/McQuade-Chem-ML/blob/39ddeee175d1eeb889c2b5440484f165efbc8b45/main.py#L87): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (158 > 79 characters)

qle2 commented on core/features.py:
```diff
@@ -20,7 +20,7 @@ def featurize(df, model_name, num_feat=None):
 
     # Remove un-normalized feature option depending on model type
-    if model_name in ['knn', 'mlp', 'svr']:
+    if model_name in ['knn', 'mlp', 'svr', 'svc', 'knc']:
```
This is just a question that does not relate to your code: Have you checked how machine learning results differ between different normalization methods available in sklearn for your classification models?

dickeygh replied:
I have done some research into this and I have also found that these models generally perform better with the rdkit2dnormalized featurization option (some won't run properly with rdkit2d).

qle2 commented on core/Classifiers.py:
```diff
@@ -0,0 +1,14 @@
+from sklearn.svm import SVC
```
Nothing wrong with the code. I suggest renaming your script to `classifiers.py`. Just a quality-of-live kind of improvement.

Dlux804 replied:
I agree with Quang.  Use the lower case, please.

qle2 commented on main.py:
```diff
@@ -12,66 +12,90 @@ def main():
-        else:
-            feats = featurize
+    # Asking the user to decide between running classification models or regression models
+    C = str(input("Enter classification for classification, regression for regression: "))
```
Generally we don't use upper case to name variables. Same reason as naming files. 

Also, typing `classification` and `regression` is very long and we are bound to make typos. Use other strings instead like `c` for classification and `r` for regression.

qle2 commented:
It's well put together overall. Just some minor naming convention stuff (This guide is pretty helpful: https://www.python.org/dev/peps/pep-0008/#naming-conventions). I like how well documented your functions are scripts are.

dickeygh commented:
I made the following changes to this pull request on 6/3:
Moved importing of classification evaluation metrics to the top of the analysis.py file.

Added an extra classifier (RandomForestClassifier) to classifiers.py. This also resulted in small changes on line 23 of features.py and line 21 of main.py.

Changed classifiers.py to be lowercase.

Changed the variable "c" to be lowercase in main.py. Changed the input statement to prompt for c and r instead of classification and regression. (Line 16)

Changed line 17 of name.py to work across different machines.

qle2 commented on core/models.py:
```diff
@@ -2,7 +2,7 @@
 '''
 
-from core import ingest, features, grid, regressors, analysis, name
+from core import ingest, features, grid, regressors, analysis, name, classifiers
```
Random Forest has a property called `feature_importances_` that will give a percentage score to all features on how important they are or how much they contribute to the learning process. We use that property to make feature importance graph in line 81 and 82. Can you please add in the string representation of Random Forest Classification on the _if_ statement in line 81?
